src/main/controllers/RemoveListConfirmationController.ts

```diff
@@ -60,17 +53,64 @@ export default class RemoveListConfirmationController {
                 break;
             }
             case 'no': {
-                res.redirect(`/remove-list-search-results?locationId=${formData.locationId}`);
+                res.clearCookie('formCookie');
+                res.redirect(`/remove-list-search-results?locationId=${locationId}`);
                 break;
             }
             default:
-                artefact.listTypeName = manualUploadService.getListItemName(artefact.listType);
+                if (Array.isArray(listsToDelete)) {
+                    for (const artefactId of listsToDelete) {
+                        await addListDetailsToArray(artefactId, req.user?.['userId'], listData);
+                    }
+                } else {
+                    await addListDetailsToArray(listsToDelete, req.user?.['userId'], listData);
+                }
                 res.render('remove-list-confirmation', {
                     ...cloneDeep(req.i18n.getDataByLanguage(req.lng)['remove-list-confirmation']),
-                    artefact,
-                    court: await courtService.getLocationById(formData.locationId),
+                    court: await courtService.getLocationById(locationId),
+                    removalList: manualUploadService.formatListRemovalValues(listData),
                     displayError: true,
                 });
         }
     }
+
```

Comment:
Can we move the below methods into a service class instead? It will keep the controller / service layer split clear + also slightly easier to test